addons/ai_apps/static/src/message_patch.js

```diff
@@ -0,0 +1,50 @@
+import { useInsertActionOnly } from "./message_actions";
+
+import { Message } from "@mail/core/common/message";
+import { MessageActionMenuMobile } from "@mail/core/common/message_action_menu_mobile";
+import { patch } from "@web/core/utils/patch";
+import { isMobileOS } from "@web/core/browser/feature_detection";
+
+
+patch(Message.prototype, {
+    setup() {
+        super.setup();
+        if (this.props.thread?.channel_type === "ai_composer") {
+            this.messageActions = useInsertActionOnly();
+        }
+    },
+    openMobileActions(ev) {
+        if (this.props.thread.channel_type !== "ai_composer") {
+            super.openMobileActions(ev);
+        }
+        if (!isMobileOS()) {
+            return;
+        }
+        ev?.stopPropagation();
+        this.state.actionMenuMobileOpen = true;
+        this.dialog.add(
+            MessageActionMenuMobile,
+            {
+                message: this.props.message,
+                thread: this.props.thread,
+                isFirstMessage: this.props.isFirstMessage,
+                messageToReplyTo: this.props.messageToReplyTo,
+                openReactionMenu: () => this.openReactionMenu(),
+                state: this.state,
+                insertToEditor: () => this.env.insertToEditor(),
+            },
+            { context: this, onClose: () => (this.state.actionMenuMobileOpen = false) }
+        );
+    },
```

Comment:
This is copy-paste just for the `insertToEditor`. With previous comment, this shouldn't be necessary, and you can completely avoid the copy-paste.

By the way, just to show how problematic this is to copy-paste: recently I removed `messageToReplyTo` to clean the code. Copy-paste makes hard to keep code short and simple.